### PR TITLE
more pedantic silencing

### PR DESCRIPTION
**Pull request by arrufat**
Sorry, I didn't catch these with my previous commit (now I've rebuilt from scratch all the examples)

### CODE REVIEW

**Comment by arrufat** [comment]
I'm not sure about this second commit, if you don't like it, I will revert it.

File: dlib/dnn/layers.h [hunk 1]
```diff
@@ -2241,12 +2241,14 @@ namespace dlib
             if (version != "add_prev_")
                 throw serialization_error("Unexpected version '"+version+"' found while deserializing dlib::add_prev_.");
         }
-
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wunused-parameter"
```
Comment (davisking) [inline dlib/dnn/layers.h]: This is a pretty heavy solution.  How about just saying `friend std::ostream& operator<<(std::ostream& out, const add_prev_& )`?

Comment (arrufat) [reply dlib/dnn/layers.h]: Yes, you're right, that works better :) thank you!

**Comment by davisking** [comment]
Looks good, thanks for another PR :)

**Comment by arrufat** [comment]
my pleasure :)

**Pull request merged**